packages/web/components/drawers/token-select-drawer.tsx

```diff
@@ -332,107 +334,118 @@ export const TokenSelectDrawer: FunctionComponent<{
               <Spinner className="m-auto" />
             ) : (
               <div className="flex flex-col overflow-auto">
-                {assets.map(
-                  (
-                    {
-                      coinDenom,
-                      coinMinimalDenom,
-                      coinImageUrl,
-                      coinName,
-                      amount,
-                      usdValue,
-                      isVerified,
-                    },
-                    index
-                  ) => {
-                    return (
-                      <button
-                        key={coinMinimalDenom}
-                        className={classNames(
-                          "flex cursor-pointer items-center justify-between py-2 px-5",
-                          "transition-colors duration-150 ease-out",
-                          {
-                            "bg-osmoverse-900": keyboardSelectedIndex === index,
-                          }
-                        )}
-                        data-testid="token-select-asset"
-                        onClick={(e) => {
-                          e.stopPropagation();
-                          onClickAsset?.(coinDenom);
-                        }}
-                        onMouseOver={() => setKeyboardSelectedIndex(index)}
-                        onFocus={() => setKeyboardSelectedIndex(index)}
-                        {...{
-                          [dataAttributeName]: getTokenItemId(uniqueId, index),
-                        }}
-                      >
-                        <div
+                {assets
+                  // should not be able to select the same token for to / from
+                  .filter((asset) =>
+                    isFromSelect
+                      ? asset.amount?.denom !== swapState.toAsset?.coinDenom
+                      : asset.amount?.denom !== swapState.fromAsset?.coinDenom
```

Comment:
Were there issues with the drawer just selecting the same to/from token included in the selectable token array? Then the drawer would close. IMO I don't see an issue with the user re-selecting an asset.